src/Symfony/Component/Asset/Preload/HttpFoundationPreloadManager.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Asset\Preload;
+
+use Symfony\Component\Asset\Exception\InvalidArgumentException;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * Preload manager for the HttpFoundation component.
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ */
+class HttpFoundationPreloadManager implements PreloadManagerInterface
+{
+    private $resources = array();
```

Comment:
No it's not possible because this state gather all assets to add to the link.
However I can add a new method to clear it that will be called in the listener to avoid the memory leak.